wolfssl/internal.h

```diff
@@ -4241,8 +4263,90 @@ typedef enum EarlyDataState {
 } EarlyDataState;
 #endif
 
-/* wolfSSL ssl type */
-struct WOLFSSL {
+#ifdef WOLFSSL_DTLS13
+
+typedef struct Dtls13UnifiedHdrInfo {
+    word16 recordLength;
+    word16 headerLength;
+    byte seqLo;
+    byte seqHi;
+    byte seqHiPresent:1;
+    byte epochBits;
+} Dtls13UnifiedHdrInfo;
+
+enum  {
+    DTLS13_EPOCH_EARLYDATA = 1,
+    DTLS13_EPOCH_HANDSHAKE = 2,
+    DTLS13_EPOCH_TRAFFIC0 = 3
+};
+
+typedef struct Dtls13Epoch {
+    word32 epochNumber;
+
+    word16 nextSeqNumberHi;
+    word32 nextSeqNumberLo;
+
+    word16 nextPeerSeqNumberHi;
+    word32 nextPeerSeqNumberLo;
+
+    word32 window[WOLFSSL_DTLS_WINDOW_WORDS];
+
+    /* key material for the epoch */
+    byte client_write_key[MAX_SYM_KEY_SIZE];
+    byte server_write_key[MAX_SYM_KEY_SIZE];
+    byte client_write_IV[MAX_WRITE_IV_SZ];
+    byte server_write_IV[MAX_WRITE_IV_SZ];
+
+    byte aead_exp_IV[AEAD_MAX_EXP_SZ];
+    byte aead_enc_imp_IV[AEAD_MAX_IMP_SZ];
+    byte aead_dec_imp_IV[AEAD_MAX_IMP_SZ];
+
+    byte client_sn_key[MAX_SYM_KEY_SIZE];
+    byte server_sn_key[MAX_SYM_KEY_SIZE];
+
+    byte isValid;
+} Dtls13Epoch;
+
+#define DTLS13_EPOCH_SIZE 4
+
+#define DTLS13_RETRANS_RN_SIZE 3
```

Comment:
Use this template so that this can be customized with compile time flags
```
#ifndef DTLS13_RETRANS_RN_SIZE
#define DTLS13_RETRANS_RN_SIZE 3
#endif
```